tests/integration/conftest.py

```diff
@@ -69,3 +69,23 @@ def local_pypiserver() -> Iterator[None]:
             raise RuntimeError("pypiserver didn't start fast enough")
 
         yield
+
+
+def does_not_test_any_affected_pms(test: Any, affected_pms_to_test: set[str]) -> bool:
+    pm_under_test = Path(test.fspath).stem.lstrip("test_")
+    return pm_under_test not in affected_pms_to_test
+
+
+def pytest_collection_modifyitems(session: Any, config: Any, items: Any) -> None:
+    skip_mark = pytest.mark.skip(reason="No changes to tested code")
+    to_test = utils.determine_integration_tests_to_run_for("HEAD")
+    if to_test is utils.TEST_EVERYTHING:
+        return
+    elif to_test is utils.TEST_NOTHING:
```

Comment:
This is a compound if-elif-else block, it visually makes more sense the way it's written despite a return being used.

Reply:
I would disagree it is visually any better. By the way it is reported by [ruff](https://docs.astral.sh/ruff/rules/superfluous-else-return/) and [pylint](https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/no-else-return.html) as well.

I imagine something like that to be the best, but I don't know if it would be possible (just an idea).
```
def pytest_collection_modifyitems(session: Any, config: Any, items: Any) -> None:
    skip_mark = pytest.mark.skip(reason="No changes to tested code")
    to_be_skipped = utils.determine_integration_tests_to_be_skip("HEAD")

    for test in to_be_skipped:
        test.add_marker(skip_mark)
```

Reply:
I do agree from reader's POV the collector function reads easier knowing that all the selection logic on which tests to run is abstracted in a helper. So to me it's just moving code around, the logic needs to live somewhere, either in the collector (partially) or the helper (completely).

Reply:
> I tried returning a single iterable with all things to switch off, but did not like how it looked.

What was the problem? I still feel like the logic around this could be simplified.

Reply:
The problem is with determining which tests not to skip. In my approach I effectively return markers for types of tests not to skip, my helpers know nothing about tests themselves, just about different types of code and which marker corresponds to which changed part. This comes at a price of teaching pytest about these markers. The alternative approach would be to implement test names accumulation in helpers, which would make the helpers more complex, thus more error-prone and would also result in re-implementing parts of pytest (i.e. tests collection). This is a more complicated approach, I tried sketching it first, did not like the complexity and opted for the solution which I posted.

Reply:
Thanks for the snippet @slimreaper35 , I did indeed overthink this. I will apply the change to the next revision.